### PR TITLE
[tests] Add Utils singleton

### PR DESCRIPTION
**Pull request by townsend2010**
This drops the previous premock method for mocking utils functions.

### CODE REVIEW

**Review by ricab — COMMENTED** [review]
Nice and very useful, thanks! 

I have one main point about the mock plus a couple of questions.

File: src/daemon/CMakeLists.txt [hunk 1]
```diff
@@ -61,9 +61,6 @@ target_link_libraries(delayed_shutdown
 )
 
 add_daemon_target(daemon)
```
[the remark this answers is not in the document]
Comment (townsend2010) [reply src/daemon/CMakeLists.txt:63]: Oh, right, I should just remove all of that too.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -104,13 +103,10 @@ target_compile_definitions(sftp_test PRIVATE
   ${c_mock_defines})
 target_compile_definitions(ssh_client_test PRIVATE
   ${c_mock_defines})
-target_compile_definitions(daemon_test PRIVATE
-  filesystem_bytes_available=ut_filesystem_bytes_available)
 
 target_link_libraries(multipass_tests
   cert
   client
-  daemon_test
```
[the remark this answers is not in the document]
Comment (townsend2010) [reply tests/CMakeLists.txt:113]: Yeah, you're probably right, but somehow it still works???? :man_shrugging:

File: tests/mock_utils.h [hunk 3]
```diff
@@ -15,18 +15,24 @@
...
+class MockUtils : public Utils
 {
-DECL_MOCK(filesystem_bytes_available);
-} // namespace utils
-} // namespace multipass
+public:
+    using Utils::Utils;
+    MOCK_METHOD1(filesystem_bytes_available, qint64(const QString&));
+
+    MP_MOCK_SINGLETON_BOILERPLATE(::testing::NiceMock<MockUtils>, Utils);
```
[the remark this answers is not in the document]
Comment (ricab) [reply tests/mock_utils.h:35]: As discussed in the chat, the correct thing to do would be to allow specifying the desired mock character when using and not in the class declaration (e.g. `StrictMock<MockUtils>::inject()`). Unfortunately that wouldn't work without more messing with singleton mocks. 

Furthermore, as Chris pointed out, existing singleton mocks register as `NiceMocks` too: `Settings` and `StandardPaths`. And the later does defer to the base implementation by default, thereby relying on the real filesystem (especially for `locate`).

So, the most reasonable is to stick to the same standards here. Changes could be applied to all such cases separately.

File: tests/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (townsend2010) [reply tests/CMakeLists.txt:113]: I guess something else is linking it, but it's safer to be explicit.  Fixed.

File: src/daemon/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (townsend2010) [reply src/daemon/CMakeLists.txt]: Ok, all removed.

**Review by ricab — APPROVED** [review]
Alright, LGTM. Thanks again!

bors r+

**Comment by ricab** [comment]
bors retry

**Comment by townsend2010** [comment]
bors retry

**Pull request merged**